astropy/units/utils.py

```diff
@@ -192,9 +193,11 @@ def is_effectively_unity(value: Complex) -> bool:
         )
 
 
-def sanitize_scale(scale: Complex) -> Complex:
+def sanitize_scale(scale: Complex) -> UnitScale:
     if is_effectively_unity(scale):
         return 1.0
+    if not scale:
```

Comment:
Perhaps it would be better to check
```suggestion
    if is_effectively_unity(scale + 1):
```

Reply:
So, in my suggestion, this part would move to `verify_scale`, as would the dtype check. Essentially:
```
def verify_scale(scale):
    if not scale:
        raise UnitScaleError("cannot create a unit with a scale of 0.")
    return scale if scale.__class__ is float or not hasattr(scale, "dtype") else scale.item()
```
I *think* it is not necessary to check effective unity, etc., as that is done later anyway.